src/core/lwt_list.ml

```diff
@@ -30,13 +30,9 @@ let rec iter_s f l =
         f x >>= fun () ->
         iter_s f l
 
-let rec iter_p f l =
-  match l with
-    | [] ->
-        Lwt.return_unit
-    | x :: l ->
-        let tx = f x and tl = iter_p f l in
-        tx >>= fun () -> tl
+let iter_p f l =
+  let ts = List.map f l in
```

Comment:
[`List.map`](https://caml.inria.fr/pub/docs/manual-ocaml/libref/List.html) is unfortunately not tail-recursive (see link). The same applies to `List.mapi`. For `map`, I think you can do something like `List.rev (List.rev_map f l)`. For `mapi`, I guess a custom helper is necessary.